packages/adapter-d1/src/index.ts

```diff
@@ -0,0 +1,353 @@
+ * 
+ * The table prefix "authjs_" is not configurable at this time.
+ * 
+ * You can use something like the following to attempt the migration once each time your worker starts up.  Running migrations more than once will not erase your existing tables.
+ * ```javascript
+ * import { up } from "@next-auth/d1-adapter"
+ * 
+ * let migrated = false;
+ * async function migrationHandle({event, resolve}) {
+ *  if(!migrated) {
+ *    try {
+ *      await up(event.platform.env.db)
+ *      migrated = true
+ *    } catch(e) {
+ *      console.log(e.cause.message, e.message)
+ *    }
+ *  }
+ *  return resolve(event)
+ * }
+ * ```
+ * 
+ *
+ * You can also initialize your tables manually.  Look in [init.ts](https://github.com/nextauthjs/next-auth/packages/adapter-d1/src/migrations/init.ts) for the relevant sql.
+ * Paste and run the SQL into your D1 dashboard query tool.
+ * 
+ **/
+export function D1Adapter(db:BothDB): Adapter {
+
+  // we need to run migrations if we dont have the right tables
+
+  return {
+    async createUser(user) {
+      console.log('Creating User:', user)
+      const id: string = crypto.randomUUID();
+      const createBindings = [id, user.name , user.email, user.emailVerified?.toISOString(), user.image];
+      const getBindings = [id];
+      const newUser = await createRecord<AdapterUser>(
+        db, 
+        CREATE_USER_SQL, createBindings,
+        GET_USER_BY_ID_SQL, getBindings
+      );
+      // this method cant return null, gotta throw
+      if(newUser) return newUser;
+      throw new Error(`Couldn't create a new user`);
+    },
+    async getUser(id) {
+      return await getRecord<AdapterUser>(db,GET_USER_BY_ID_SQL,[id]);
+    },
+    async getUserByEmail(email) {
+      return await getRecord<AdapterUser>(db,GET_USER_BY_EMAIL_SQL,[email]);
+    },
+    async getUserByAccount({ providerAccountId, provider }) {
+      return await getRecord<AdapterUser>(db,GET_USER_BY_ACCOUNTL_SQL,[providerAccountId,provider]);
+      
+    },
+    async updateUser(user) {
+      const params = await getRecord<AdapterUser>(db,GET_USER_BY_ID_SQL,[user.id]);
+      if(params) {
+        // copy any properties not in the update into the existing one and use that for bind params
+        // covers the scenario where the user arg doesnt have all of the current users properties
+        Object.assign(params,user);
+        const res = await updateRecord(db,UPDATE_USER_BY_ID_SQL,[params.name, params.email, params.emailVerified?.toISOString(), params.image, params.id]);
+        if(res.success) {
+         // we could probably just return 
+         const user = await getRecord<AdapterUser>(db,GET_USER_BY_ID_SQL,[params.id]);
+         if(user) return user;
+         throw new Error(`couldnt find user after update with id ${params.id}`);
+        } 
+        throw new Error(`couldnt update user with data ${JSON.stringify(user)}`);
+      }
+      throw new Error(`couldnt find user to update from id ${user.id}`);
```

Comment:
I think we can remove these line, how do you think? just let the helper methods throw whatever errors is ok.